src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/IPackageFile.cs

```diff
@@ -22,21 +23,31 @@ string Path
         /// Gets the path that excludes the root folder (content/lib/tools) and framework folder (if present).
         /// </summary>
         /// <example>
-        /// If a package has the Path as 'content\[net40]\scripts\jQuery.js', the EffectivePath 
+        /// If a package has the Path as 'content\[net40]\scripts\jQuery.js', the EffectivePath
         /// will be 'scripts\jQuery.js'.
-        /// 
+        ///
         /// If it is 'tools\init.ps1', the EffectivePath will be 'init.ps1'.
         /// </example>
         string EffectivePath
         {
             get;
         }
 
+        /// <summary>
+        /// FrameworkName object representing this package file's target framework. Deprecated. Must be null on net5.0 and greater.
+        /// </summary>
         FrameworkName TargetFramework
         {
             get;
         }
 
+        /// <summary>
+        /// NuGetFramework object representing this package file's target framework. Use this instead of TargetFramework.
+        /// </summary>
+        NuGetFramework NuGetFramework
```

Comment:
add one line of whitespace between this property and `LastWriteTime`.

Also, should we mark `TargetFramework` as obsolete?  Or is it still used in non-net5.0 pack?

Reply:
If possible we should just use NuGetFramework, cause there's an easy way to go from NuGetFramework to whatever FrameworkName could get us. 

Deprecating FrameworkName could help people working on pack avoid future issues.